src/css.js

```diff
@@ -58,13 +58,31 @@ function vendorPropName( name ) {
 	}
 }
 
+// Use font-size to convert to px
+function convertToPx( elem, value ) {
+	var fontSize = jQuery.css( elem, "font-size" );
+	elem.style.fontSize = value;
+	value = parseFloat( jQuery.css( elem, "font-size" ) );
+	elem.style.fontSize = fontSize;
+	return value;
+}
+
 function setPositiveNumber( elem, value, subtract ) {
-	var matches = rnumsplit.exec( value );
-	return matches ?
+	var unit,
+		matches = rnumsplit.exec( value );
+	if ( matches ) {
+		unit = matches[ 2 ] || "px";
+
+		// Only need to convert if we're manipulating the value
+		if ( unit !== "px" && subtract ) {
+			matches[ 1 ] = convertToPx( elem, value );
```

Comment:
I've made it so we reuse matches when possible. Also, removed `rnumsplit`. `rcssNum` seems like enough.